plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResource.java

```diff
@@ -2756,6 +2758,32 @@ protected KVMStoragePoolManager getPoolManager() {
         return _storagePoolMgr;
     }
 
+    public void detachAndAttachConfigDriveISO(final Connect conn, final String vmName) {
+        // detach and re-attach configdrive ISO
+        List<DiskDef> disks = getDisks(conn, vmName);
+        DiskDef configdrive = null;
+        for (DiskDef disk : disks) {
+            if (disk.getDeviceType() == DiskDef.DeviceType.CDROM && disk.getDiskLabel() == CONFIG_DRIVE_ISO_DISK_LABEL) {
+                configdrive = disk;
+            }
+        }
+        if (configdrive != null) {
```

Comment:
We could invert the logic here to reduce block complexity.

Reply:
I like your thinking @GutoVeronezi , but in this case that would leave a return in the middle of the method. I'd rather extract the block into an extra method.